### PR TITLE
add task form popout to gantt chart

### PR DESCRIPTION
**Pull request by PavelRubis**

### CODE REVIEW

File: src/GanttElastic.vue [hunk 1]
```diff
@@ -1181,8 +1183,9 @@ const GanttElastic = {
     /**
      * click event handler
      */
-    onClick() {
-      if (this.state.refs.chartDependencyPopup) {
+    onClick(event) {
+      const popup = this.state.refs.dependencyPopupEl;
+      if (popup && (event.target !== popup || !popup.contains(event.target))) {
```
Comment (stasguryev) [inline src/GanttElastic.vue]: popup && !popup.contains(event.target)

File: src/components/Chart/DependencyPopup.vue [hunk 2]
```diff
@@ -136,41 +122,20 @@ export default {
...
-        case 'endToStart':
-        default:
-          startX = fromTask.xP + fromTask.widthP;
-          stopX = toTask.xP;
-          break;
-      }
       const popupWidth = this.root.state.options.dependencyPopup.width;
-      this.position.left = startX + (stopX - startX) / 2 - popupWidth / 2;
-      this.position.top = Math.max(fromTask.yP, toTask.yP) + this.root.state.options.row.height;
+      const depLineRect = this.info.dependencyLineEl.getClientRects()[0];
```
Comment (stasguryev) [inline src/components/Chart/DependencyPopup.vue]: getboundingclientrect?

File: src/components/Chart/Row/Task.mixin.js [hunk 3]
```diff
@@ -47,6 +47,18 @@ export default {
...
+    },
+    /**
+     * Handler for click event
+     *
+     * @param {Event} event
+     */
+    onBarClick(event) {
+      const callBacks = this.root.state.options.callBacks;
+      if (callBacks && callBacks.onTaskClick) {
+        callBacks.onTaskClick(this.$refs.taskBar, { id: this.task.id, title: this.task.title });
```
Comment (stasguryev) [inline src/components/Chart/Row/Task.mixin.js]: onTaskClick( options ) ?

File: src/GanttElastic.vue (hunk 1 above)
Comment (PavelRubis) [reply src/GanttElastic.vue]: done

File: src/components/Chart/DependencyPopup.vue (hunk 2 above)
Comment (PavelRubis) [reply src/components/Chart/DependencyPopup.vue]: done

File: src/components/Chart/Row/Task.mixin.js (hunk 3 above)
Comment (PavelRubis) [reply src/components/Chart/Row/Task.mixin.js]: done

**Pull request merged**